plugin/audit_log/audit_log.c

```diff
@@ -468,8 +488,31 @@ char *audit_log_general_record(char *buf, size_t buflen,
                      "\"%s\",\"%s\",\"%s\",\"%s\",\"%lu\",%d,\"%s\",\"%s\","
                      "\"%s\",\"%s\",\"%s\",\"%s\"\n" };
 
-  query= escape_string(event->general_query, event->general_query_length,
-                       endptr, endbuf - endptr, &endptr, &full_outlen);
+  query_length= system_charset_info->mbmaxlen * event->general_query_length;
+
+  if (query_length < (size_t) (endbuf - endptr))
+  {
+    uint errors;
+    query_length= my_convert(endptr, query_length, system_charset_info,
+                             event->general_query,
+                             event->general_query_length,
+                             event->general_charset, &errors);
```

Comment:
UTF-8 can represent all the other character sets, right? Should then be asserted that errors == 0?